### PR TITLE
Move convert from bytecode instruction to primitive call

### PR DESCRIPTION
**Pull request by hmaurer**

### CODE REVIEW

**Comment by vendethiel** [comment]
Don't merge without approval :|

File: examples/call-return.undo-bc [hunk 1]
```diff
@@ -4,7 +4,9 @@ ret val
 
 $label to_string
 # obviously dispatches based on type here *kof*...
-convert int str
+push str str
+push str int
+call primitive convert
```
Comment (vendethiel) [inline examples/call-return.undo-bc:9]: Operands are very important

Comment (vendethiel) [inline examples/call-return.undo-bc:9]: `call primitive int->str` would be okay, id est

File: src/smallstack.rs [hunk 2]
```diff
@@ -113,6 +104,24 @@ impl<'a> VM<'a> {
             None => panic!("VM error: no argument supplied to `typeof`"),
         },
 
+        &["call", "primitive", "convert"] => if let
```
Comment (vendethiel) [inline src/smallstack.rs:107]: Please don't encode instruction toggles as stack arguments

Comment (a3gis) [inline src/smallstack.rs]: Necessary to allow this: https://github.com/a3gis/smallstack-js/blob/master/examples/prime.js#L28

Comment (hmaurer) [inline src/smallstack.rs:107]: Yeah, fixed by https://github.com/vendethiel/smallstack/pull/12/commits/d09e686077264651a35eec71e71134a461f54c20.

File: examples/call-return.undo-bc (hunk 1 above)
Comment (hmaurer) [inline examples/call-return.undo-bc:9]: c.f. https://github.com/vendethiel/smallstack/pull/12/commits/d09e686077264651a35eec71e71134a461f54c20

**Pull request merged**